databuilder/README.md

```diff
@@ -1700,6 +1700,22 @@ Adds the same set of tags to all tables produced by the job.
 #### [GenericTransformer](./databuilder/transformer/generic_transformer.py)
 Transforms dictionary based on callback function that user provides.
 
+#### [ComplexTypeTransformer](./databuilder/transformer/complex_type_transformer.py)
```

Comment:
perhaps:
1. explain how the is_terminal_type stuff works
2. explain how Scalar is actually the fallback
3. explain what is `kind`
4. explain any non-obvious relationships with the frontend. The person implementing this will be some backend person who won't want to figure out on their own that, for example, if the kind is array or map, it gets a little narrow row, etc.